flax/nn/__init__.py

```diff
@@ -29,3 +29,6 @@
 from .recurrent import LSTMCell, GRUCell, ConvLSTM, OptimizedLSTMCell
 from .stochastic import make_rng, stochastic, dropout, is_stochastic
 # pylint: enable=g-multiple-import
+import warnings
+warnings.filterwarnings("default", category=DeprecationWarning, module=__name__)
```

Comment:
What does this line do? (Can we add a comment before it?) If we don't include it does the `warning.warn` below not work?

Reply:
Deprecation warning are muted by default, this makes sure the person sees the warning
https://stackoverflow.com/a/51903782

Reply:
Can you write this as a comment in the code then, for future people looking at it?

Reply:
Done!